src/main/java/io/kestra/plugin/ai/retriever/SqlDatabaseRetriever.java

```diff
@@ -0,0 +1,231 @@
+                      modelName: gpt-4o-mini
+                      apiKey: "{{ kv('OPENAI_API_KEY') }}"
+                    contentRetrievers:
+                      - type: io.kestra.plugin.ai.retriever.SqlDatabaseRetriever
+                        jdbcUrl: "jdbc:postgresql://localhost:5432/mydb"
+                        username: "{{ kv('DB_USER') }}"
+                        password: "{{ kv('DB_PASSWORD') }}"
+                        driver: "org.postgresql.Driver"
+                        provider: "OPENAI"
+                        modelName: "gpt-4o-mini"
+                        apiKey: "{{ kv('OPENAI_API_KEY') }}"
+                    prompt: "What are the top 5 customers by revenue?"
+                """
+        )
+    }
+)
+public class SqlDatabaseRetriever extends ContentRetrieverProvider {
+
+    public enum Provider {
+        OPENAI,
+        AZURE_OPENAI,
+        GEMINI,
+        ANTHROPIC,
+        MISTRAL,
+        OLLAMA,
+        QWEN,
+        VERTEX_GEMINI,
+        VERTEX,
+        WORKERS,
+        BEDROCK,
+        LOCAL
+    }
+
+    @Schema(title = "JDBC connection URL to the target database.")
+    @NotNull
+    private Property<String> jdbcUrl;
+
+    @Schema(title = "Database username.")
+    @NotNull
+    private Property<String> username;
+
+    @Schema(title = "Database password.")
+    @NotNull
+    private Property<String> password;
+
+    @Schema(title = "Fully qualified JDBC driver class name (e.g. org.postgresql.Driver, com.mysql.cj.jdbc.Driver).")
+    @NotNull
+    private Property<String> driver;
+
+    @Schema(title = "LLM provider.")
+    @NotNull
+    private Property<Provider> provider;
+
+    @Schema(title = "Chat model name to use for SQL reasoning (e.g. gpt-4o-mini).")
+    @NotNull
+    private Property<String> modelName;
+
+    @Schema(title = "API key for the LLM provider.")
+    @NotNull
+    private Property<String> apiKey;
+
+    @Override
+    public ContentRetriever contentRetriever(RunContext runContext) throws IllegalVariableEvaluationException {
+        String rJdbcUrl = runContext.render(this.jdbcUrl).as(String.class).orElseThrow();
+        String rUsername = runContext.render(this.username).as(String.class).orElseThrow();
+        String rPassword = runContext.render(this.password).as(String.class).orElseThrow();
+        String rDriver = runContext.render(this.driver).as(String.class).orElseThrow();
+        Provider rProvider = runContext.render(this.provider).as(Provider.class).orElseThrow();
+        String rModelName = runContext.render(this.modelName).as(String.class).orElseThrow();
+        String rApiKey = runContext.render(this.apiKey).as(String.class).orElseThrow();
+
+        try {
+            Class.forName(rDriver);
+        } catch (ClassNotFoundException e) {
+            throw new IllegalArgumentException("JDBC driver not found: " + rDriver, e);
+        }
+
+        DataSource dataSource = new DataSource() {
```

Comment:
This is not ideal.
Should it not be configurable as for the driver?

Or should we not add an enum for supported databases? In fact, the issue here is that the JDBC driver must be inside the classpath, and today, we have three: h2, MySQL, and PostgreSQL.
Some users may know how to extend the classpath, but most would not be able to use this with any other JDBC driver.

The other solution would be to use a datasource impl we have at hand, for ex Hikari's DriverDataSource.
Hikari should be added as a compileOnly dependency and testImplementation dependency as we have it inside Kestra core.